src/Microsoft.AspNetCore.Mvc.ApiExplorer/ProblemDetailsApiDescriptionProvider.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+
+namespace Microsoft.AspNetCore.Mvc.ApiExplorer
+{
+    public class ProblemDetailsApiDescriptionProvider : IApiDescriptionProvider
+    {
+        private readonly IModelMetadataProvider _modelMetadaProvider;
+
+        public ProblemDetailsApiDescriptionProvider(IModelMetadataProvider modelMetadataProvider)
+        {
+            _modelMetadaProvider = modelMetadataProvider;
+        }
+
+        /// <remarks>
+        /// The order is set to execute after the <see cref="DefaultApiDescriptionProvider"/>.
+        /// </remarks>
+        public int Order => -1000 + 10;
+
+        public void OnProvidersExecuted(ApiDescriptionProviderContext context)
+        {
+        }
+
+        public void OnProvidersExecuting(ApiDescriptionProviderContext context)
+        {
+            foreach (var apiDescription in context.Results)
+            {
+                if (!apiDescription.ActionDescriptor.FilterDescriptors.Any(f => f.Filter is ProblemDetailsAttribute))
+                {
+                    continue;
+                }
+
+                var parameters = apiDescription.ActionDescriptor.Parameters.Concat(apiDescription.ActionDescriptor.BoundProperties);
+                if (parameters.Any())
+                {
+                    apiDescription.SupportedResponseTypes.Add(CreateProblemResponse(StatusCodes.Status400BadRequest));
+
+                    if (parameters.Any(p => p.Name.EndsWith("id", StringComparison.OrdinalIgnoreCase)))
+                    {
+                        apiDescription.SupportedResponseTypes.Add(CreateProblemResponse(StatusCodes.Status404NotFound));
+                    }
+                }
+
+                apiDescription.SupportedResponseTypes.Add(CreateProblemResponse(statusCode: 0));
```

Comment:
I'll file a bug for this. We need a way to represent default response types

Reply:
Can you please add a comment and explain what this is?